### PR TITLE
Upgrade to AzureTRE v0.8.0

### LINKED ISSUE #37: Align with env files consolidation
## Description

As a AzureTRE-Deployment developer,
I would like to use the latest AzureTRE version and to be aligned with the new configuration

### PR DESCRIPTION
**Pull request by LizaShak**
## What is being addressed

Fixes #37 By aligning with new configuration format. 

## How is this addressed

- Add config.sample.yaml and the schema validation
- Remove old .env files

### CODE REVIEW

File: config.sample.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+---
```
Comment (tamirkamara) [inline config.sample.yaml:1]: Weren't the older .env files were copied somehow when the devcontainer started?

**Review by tamirkamara — CHANGES_REQUESTED** [review]
This PR is waiting for the next AzureTRE release. Blocking for now.

File: config.sample.yaml (hunk 1 above)
Comment (LizaShak) [reply config.sample.yaml:1]: Fixed

File: .github/workflows/build_validation_develop.yml [hunk 2]
```diff
@@ -52,3 +82,54 @@ jobs:
           VALIDATE_BASH_EXEC: true
           VALIDATE_GITHUB_ACTIONS: true
           VALIDATE_DOCKERFILE_HADOLINT: true
+
+      - name: Docs validation
```
Comment (tamirkamara) [inline .github/workflows/build_validation_develop.yml:86]: remove

File: .github/workflows/build_validation_develop.yml [hunk 3]
```diff
@@ -52,3 +82,54 @@ jobs:
...
+        if: ${{ steps.filter.outputs.docs == 'true' }}
+        run: |
+          pip install -r docs/requirements.txt
+          mkdocs build --strict
+
+      - name: Workspace Tags
+        if: ${{ steps.filter.outputs.terraform_workspaces == 'true' }}
```
Comment (tamirkamara) [inline .github/workflows/build_validation_develop.yml:93]: remove all if statements about the filter

File: .github/workflows/build_validation_develop.yml [hunk 4]
```diff
@@ -24,22 +24,52 @@ jobs:
           fetch-depth: 0
           persist-credentials: false
 
+      - uses: dorny/paths-filter@v2
```
Comment (tamirkamara) [inline .github/workflows/build_validation_develop.yml:27]: remove

File: .devcontainer/scripts/post-create.sh [hunk 5]
```diff
@@ -9,9 +9,5 @@ set -o nounset
 rm -fr AzureTRE || true
 ln -s "${AZURETRE_HOME}" AzureTRE
```
Comment (tamirkamara) [inline .devcontainer/scripts/post-create.sh:11]: copy configs

File: .github/workflows/deploy_tre_reusable.yml [hunk 6]
```diff
@@ -129,6 +132,80 @@ jobs:
...
+          if [ "${{ secrets.TRE_ID }}" == '' ]; then
+            echo "Missing secret: TRE_ID" && exit 1
+          fi
+          if [ "${{ secrets.AZURE_CREDENTIALS }}" == '' ]; then
+            echo "Missing secret: AZURE_CREDENTIALS" && exit 1
+          fi
+
+      - name: Report check status start
+        if: inputs.prHeadSha != ''
+        uses: LouisBrunner/checks-action@v1.5.0
```
Comment (tamirkamara) [inline .github/workflows/deploy_tre_reusable.yml:201]: If this action isn't already used at the very end of the workflow, then we don't need it here also.

File: config_schema.json [hunk 7]
```diff
@@ -0,0 +1,158 @@
+{
```
Comment (tamirkamara) [inline config_schema.json:1]: try to remove the file and see if the make commands still work since they use the file from the oss folder

**Review by tamirkamara — APPROVED** [review]

**Pull request merged**